charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -110,3 +110,56 @@ Return if ingress is stable.
 {{- define "ingress.isStable" -}}
   {{- eq (include "ingress.apiVersion" .) "networking.k8s.io/v1" -}}
 {{- end -}}
+
+
+{{- define "opentelemetry-collector.daemonset-podAnnotations" -}}
+{{- if eq .Values.mode "daemonset" }}
+  {{- if .Values.podAnnotations }}
```

Comment:
Why do we have this if condition here but don't have it below after `{- else if .Values.agentCollector.podAnnotations }}`?

Reply:
We probably don't need it since `.Values.podAnnotations` and `.Values.agentCollector.podAnnotations` both have the same `{}` default?

Reply:
The goal was that if mode was set to daemonset then all agentCollector settings would be ignored, even if they were set and even if .Values.podAnnotations was not set. Hence the funky outer if statement checking for daemonset.

Reply:
Oh I see what you mean. I would agree, we can send the empty {} to toYaml safely. Good thinking, I'll update that tomorrow.

Reply:
Since `podLabels` and `podAnnotations` are being merge with an existing label (`component`) and an existing annotation (`checksum/config`) I don’t think we can safely pipe {} to toYaml in those helpers.  It results in something like
```yaml
labels:
  component: standalone-collector
  {}
```
which then blows up in helm.

Reply:
Could move all the default podLabels/podAnnotations into the template and then merge them together before sending it to `toYaml`, but I think that template would get messy bc the defaults would have to be put in a dict instead of just written in the file like they are now.

Reply:
Actually I was wrong initially. For some reason I read `{{- else if .Values.agentCollector.podAnnotations }}` as `else`. Sorry for bringing the confusion. It's all done correctly.